backend/src/main/java/com/bfwg/service/impl/AuthorityServiceImpl.java

```diff
@@ -1,34 +1,42 @@
-package com.bfwg.service.impl;
-
-import java.util.ArrayList;
-import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
-import com.bfwg.model.Authority;
-import com.bfwg.repository.AuthorityRepository;
-import com.bfwg.service.AuthorityService;
-import java.util.Optional;
-
-@Service
-public class AuthorityServiceImpl implements AuthorityService {
-
-    @Autowired
-    private AuthorityRepository authorityRepository;
-
-    @Override
-    public List<Authority> findById(Long id) {
-        Optional<Authority> optauth = authorityRepository.findById(id);
-        List<Authority> auths = new ArrayList<>();
-        auths.add(optauth.get());
-        return auths;
-    }
-
-    @Override
-    public List<Authority> findByname(String name) {
-        Authority auth = authorityRepository.findByName(name);
-        List<Authority> auths = new ArrayList<>();
-        auths.add(auth);
-        return auths;
-    }
-
-}
+package com.bfwg.service.impl;
+
+import com.bfwg.model.Authority;
+import com.bfwg.model.UserRoleName;
+import com.bfwg.repository.AuthorityRepository;
+import com.bfwg.service.AuthorityService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class AuthorityServiceImpl implements AuthorityService {
+
+    private final AuthorityRepository authorityRepository;
+
+    @Autowired
+    public AuthorityServiceImpl(AuthorityRepository authorityRepository) {
+        this.authorityRepository = authorityRepository;
+    }
+
+    @Override
+    public List<Authority> findById(Long id) {
+        // TODO Auto-generated method stub
+
+        Authority auth = this.authorityRepository.getOne(id);
+        List<Authority> auths = new ArrayList<>();
+        auths.add(auth);
+        return auths;
+    }
+
+    @Override
+    public List<Authority> findByName(UserRoleName name) {
+        // TODO Auto-generated method stub
```

Comment:
This one as well. Can we remove please?

Reply:
Done

Reply:
I understand this is a huge pull request. Which is not the best way ...
But I had so many changes is this branch that I forgot ... long story, short, I still pushed the pull request button.
Take your time, no worries